File: java/250511_S5_14405.java
/**
 * https://www.acmicpc.net/problem/14405
 */
import java.io.*;
import java.util.*;
import java.util.regex.*;

class Main {
	public static void main(String[] args) throws IOException {
		try (Scanner scanner = new Scanner(System.in)) {
			String inp = scanner.next();
			Pattern pattern = Pattern.compile("^(pi|ka|chu)+$");
			System.out.println(
				pattern.matcher(inp).matches() ? "YES" : "NO"
			);
		}
	}
}

File: python/250511_S5_14405.py
# https://www.acmicpc.net/problem/14405
import re
pattern = re.compile('^(pi|ka|chu)+$')

if __name__ == "__main__":
    S = input()
    match_result = "YES" if pattern.fullmatch(S) else "NO"
    print(match_result)
